drivers/bh1900nux/bh1900nux.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 Nalys
+ *  
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_bh1900nux
+ * @{
+ *
+ * @file
+ * @brief       BH1900NUX temperature sensor driver implementation
+ *
+ * @author      Wouter Symons <wsymons@nalys-group.com> 
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "bh1900nux.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+int bh1900nux_init(bh1900nux_t *dev, const bh1900nux_params_t *params)
+{
+    if(dev == NULL | params == NULL) {
+        return -1;
+    }
+
+    /* initialize the device descriptor */
+    dev->i2c = params->i2c;
+    dev->addr = params->addr;
+
+    return BH1900NUX_OK;
+}
+
+int bh1900nux_read(const bh1900nux_t *dev, double *temp)
```

Comment:
Don't use floats if they can be avoided.
The device has a resolution of 1/16 °C, so just return mC here if you don't want to return the raw value as (16 * °C)